js/components/dateRange.js

```diff
@@ -0,0 +1,280 @@
+Fliplet.FormBuilder.field('dateRange', {
+  name: 'Date range',
+  category: 'Text inputs',
+  props: {
+    value: {
+      type: Object,
+      default: {
+        start: '',
+        end: ''
+      }
+    },
+    description: {
+      type: String
+    },
+    autofill: {
+      type: String,
+      default: 'always'
+    },
+    defaultSource: {
+      type: String,
+      default: 'submission'
+    },
+    defaultEndDays: {
+      type: Number,
+      default: 2
+    },
+    showPredefinedRanges: {
+      type: Boolean,
+      default: true
+    },
+    placeholder: {
+      type: String,
+      default: function() {
+        return '-- ' + T('widgets.form.dateRange.rangePlaceholder');
+      }
+    }
+  },
+  data: function() {
+    return {
+      dateRange: null,
+      isInputFocused: false,
+      isPreview: Fliplet.Env.get('preview'),
+      today: moment().locale('en').format('YYYY-MM-DD'),
+      selectedRange: ''
+    };
+  },
+  mounted: function() {
+    this.initDaterange();
+
+    if (this.defaultValueSource !== 'default') {
+      this.setValueFromDefaultSettings({
+        source: this.defaultValueSource,
+        key: this.defaultValueKey
+      });
+    }
+
+    if (this.autofill === 'default' || this.autofill === 'always') {
+      this.value = {
+        start: this.today,
+        end: this.today
+      };
+      this.empty = false;
+    }
+
+    if (this.autofill === 'empty') {
```

Comment:
Join this with the previous line.

```suggestion
    else if (this.autofill === 'empty') {
```

Or use `switch (this.autofill)`